src/org/omegat/util/StringUtil.java

```diff
@@ -511,4 +512,235 @@ public static String format(String str, Object... arguments) {
+                case 0x30DE: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF8F)); break;
+                case 0x30DF: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF90)); break;
+                case 0x30E0: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF91)); break;
+                case 0x30E1: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF92)); break;
+                case 0x30E2: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF93)); break;
+                case 0x30E4: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF94)); break;
+                case 0x30E6: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF95)); break;
+                case 0x30E8: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF96)); break;
+                case 0x30E9: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF97)); break;
+                case 0x30EA: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF98)); break;
+                case 0x30EB: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF99)); break;
+                case 0x30EC: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9A)); break;
+                case 0x30ED: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9B)); break;
+                case 0x30EF: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9C)); break;
+                case 0x30F3: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9D)); break;
+                case 0x3099: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9E)); break;
+                case 0x309A: result.append(genFullHalfMatchRegex((char)cp, (char)0xFF9F)); break;
+                // Hangul
+                case 0x3164: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFA0)); break;
+                case 0x3161: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFDA)); break;
+                case 0x3162: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFDB)); break;
+                case 0x3163: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFDC)); break;
+                // Others
+                case 0x2502: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFE8)); break;
+                case 0x2190: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFE9)); break;
+                case 0x2191: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFEA)); break;
+                case 0x2192: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFEB)); break;
+                case 0x2193: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFEC)); break;
+                case 0x25A0: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFED)); break;
+                case 0x25CB: result.append(genFullHalfMatchRegex((char)cp, (char)0xFFEE)); break;
+                // No match, add ch
+                default: result.append(codePoint2String(cp)); break;
```

Comment:
You don't need `codePoint2String`; you can use [`StringBuilder.appendCodePoint(int)`](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#appendCodePoint%28int%29).